docs/book/v3/form-element-manager.md

```diff
@@ -0,0 +1,279 @@
+
+## Create a Form Element Manager
+
+To create an instance, the form element manager requires a PSR-11 dependency container. The following examples uses the container implementation of [laminas-servicemanager](https://docs.laminas.dev/laminas-servicemanager/):
+
+```php
+$formElementManager = new Laminas\Form\FormElementManager(
+    new Laminas\ServiceManager\ServiceManager()
+);
+```
+
+## Fetch a Standard Element
+
+To get a supplied form element, use the class name of the element:
+
+```php
+$element = $formElementManager->get(Laminas\Form\Element\Select::class);
+```
+
+## Fetch and Configure a Standard Element
+
+Internally, the included factory `Laminas\Form\ElementFactory::class` for all types of elements, fieldsets and forms is used to create an element. This factory allows to configure an element, fieldset or form on creation:
+
+```php
+$element = $formElementManager->get(
+    Laminas\Form\Element\Select::class,
+    [
+        'name'    => 'rating',
+        'options' => [
+            'label'         => 'Rating',
+            'value_options' => [1, 2, 3, 4, 5],
+            ],
+        ],
+    ]
+);
+```
+
+The name for the element and the array with options are set as parameter to the constructor of the object on instantiation.
+
+```php
+public function __construct($name = null, iterable $options = []) {}
+```
+
+```php
+$element->getName(); // rating
+$element->getLabel(); // Rating
+$element->getValueOptions(); // [1, 2, 3, 4, 5]
+```
+
+## Fetch a Custom Element without Registration
+
+The form element manager allows to fetch custom elements without prior registration on the manager.
+
+The following example creates a custom element:
+
+```php
+final class ExampleElement extends Laminas\Form\Element
+{
+    // …
+}
+```
+
+The form element manager can create these custom element by the related class name:
+
+```php
+$element = $formElementManager->get(ExampleElement::class);
+```
+
+Also here, the included factory `Laminas\Form\ElementFactory::class` is used by the manager to instance the element which allows the pass the name and options like before:
+
+```php
+$element = $formElementManager->get(
+    ExampleElement::class,
+    [
+        'name'    => '…',
+        'options' => [ 
+            // …
+        ],
+    ]
+);
+```
+
+## Register and Fetch a Custom Element with a Factory
+
+If a custom element needs some preparation for creation then a factory can be used.
+
+The following example creates a class as factory for the element:
+
+```php
+final class ExampleElementFactory
+{
+    public function __invoke(Psr\Container\ContainerInterface): ExampleElement
+    {
+        $element = new ExampleElement();
+        $element->setOption('example_param', 'value');
+        
+        return $element;
+    }
+}
+```
+
+Register the custom element on the form element manager by using a factory class:
+
+```php
+$formElementManager->setFactory(
+    ExampleElement::class,
+    ExampleElementFactory::class
+);
+```
```

Comment:
I wonder if we should demonstrate adding the factory _via configuration_ as well? And/or detail that programmatic addition like this should be done in a delegator factor for the `FormElementManager` service?

Reply:
> I wonder if we should demonstrate adding the factory _via configuration_ as well?

Right, but the configuration in an application is added on a separate page. I think this can be found better via the navigation and it doesn't make these pages even longer.

Reply:
I will check another option via tabs like we already have in laminas-i18n.

https://docs.laminas.dev/laminas-i18n/filters/alnum/#using-locale

Reply:
My main point is: in both laminas-mvc and Mezzio applications, you almost NEVER programmatically add factories to the form element manager. You would instead configure them.

The one place you might programmatically add them is via a delegator factory, though, again, that's specific to the Laminas ecosystem (as it's a feature of laminas-servicemanager).

But it comes back to: we should show the most common use cases as the primary examples, and then anything manual like this as a secondary example.

Reply:
_It's nice to see that the explanations for the integration in the application are now being taken into account. 😄_ 

---

To make it clear, the basics of the manager is to be explained here and nothing more, nothing less.
The usage in laminas-mvc and Mezzio are created separately. Then links and explicit coloured notes will also be added to this page here.
And the explanations for laminas-mvc and Mezzio will get their own entry in the navigation, because these explanations are important and should be clearly visible.

> …we should show the most common use cases as the primary examples, and then anything manual like this as a secondary example.

Unfortunately, this realization comes too late, because all documentations are written exactly the other way around. And now we have to deal with it.

See also the related project: https://github.com/orgs/laminas/projects/3